storage/jellyfish-merkle/src/restore/mod.rs

```diff
@@ -689,3 +690,21 @@ where
         self.store.write_node_batch(&self.frozen_nodes)
     }
 }
+
+impl<V: crate::Value> StateSnapshotReceiver<V> for JellyfishMerkleRestore<V> {
+    fn add_chunk(
+        &mut self,
+        chunk: Vec<(HashValue, V)>,
+        proof: SparseMerkleRangeProof,
+    ) -> Result<()> {
+        self.add_chunk_impl(chunk, proof)
+    }
+
+    fn finish(self) -> Result<()> {
+        self.finish_impl()
+    }
+
+    fn finish_box(self: Box<Self>) -> Result<()> {
```

Comment:
n00b question: why do we also need the box implementation?

Reply:
`restore_box.finish()` is not allowed because the content of the box is destroyed before the box (or some similar reason, I'm not an expert either >_< )

and a blanket implementation on the trait definite doesn't work either because the size of `Self` is unknown.. 

And then there's the `Box::into_inter()` which is unstable..

Not sure if experts get a better trick. @bmwill ? :P

Reply:
Don't use trait objects (dyn T) would be my recommended trick ;)